platform/core/src/services/CineService/CineService.ts

```diff
@@ -1,82 +1,80 @@
-const name = 'CineService';
+import { PubSubService } from '../_shared/pubSubServiceInterface';
 
-const publicAPI = {
-  name,
-  getState: _getState,
-  setCine: _setCine,
-  setIsCineEnabled: _setIsCineEnabled,
-  playClip: _playClip,
-  stopClip: _stopClip,
-  onModeExit: _onModeExit,
-  setServiceImplementation,
-};
+class CineService extends PubSubService {
+  public static readonly EVENTS = {
+    CINE_STATE_CHANGED: 'event::cineStateChanged',
+  };
 
-const serviceImplementation = {
-  _getState: () => console.warn('getState() NOT IMPLEMENTED'),
-  _setCine: () => console.warn('setCine() NOT IMPLEMENTED'),
-  _playClip: () => console.warn('playClip() NOT IMPLEMENTED'),
-  _stopClip: () => console.warn('stopClip() NOT IMPLEMENTED'),
-  _setIsCineEnabled: () => console.warn('setIsCineEnabled() NOT IMPLEMENTED'),
-};
-
-function _getState() {
-  return serviceImplementation._getState();
-}
-
-function _setCine({ id, frameRate, isPlaying }) {
-  return serviceImplementation._setCine({ id, frameRate, isPlaying });
-}
+  public static REGISTRATION = {
+    name: 'cineService',
+    altName: 'CineService',
+    create: ({ configuration = {} }) => {
+      return new CineService();
+    },
+  };
 
-function _setIsCineEnabled(isCineEnabled) {
-  return serviceImplementation._setIsCineEnabled(isCineEnabled);
-}
+  serviceImplementation = {};
 
-function _playClip(element, playClipOptions) {
-  return serviceImplementation._playClip(element, playClipOptions);
-}
+  constructor() {
+    super(CineService.EVENTS);
+    this.serviceImplementation = {};
+  }
 
-function _stopClip(element) {
-  return serviceImplementation._stopClip(element);
-}
+  public getState() {
+    return this.serviceImplementation._getState();
+  }
 
-function _onModeExit() {
-  _setIsCineEnabled(false);
-}
+  public setCine({ id, frameRate, isPlaying }) {
+    return this.serviceImplementation._setCine({ id, frameRate, isPlaying });
+  }
 
-function setServiceImplementation({
-  getState: getStateImplementation,
-  setCine: setCineImplementation,
-  setIsCineEnabled: setIsCineEnabledImplementation,
-  playClip: playClipImplementation,
-  stopClip: stopClipImplementation,
-}) {
-  if (getStateImplementation) {
-    serviceImplementation._getState = getStateImplementation;
+  public setIsCineEnabled(isCineEnabled) {
+    this.serviceImplementation._setIsCineEnabled(isCineEnabled);
+    // Todo: for some reason i need to do this setTimeout since the
+    // reducer state does not get updated right away and if we publish the
+    // event and we use the cineService.getState() it will return the old state
+    setTimeout(() => {
+      this._broadcastEvent(this.EVENTS.CINE_STATE_CHANGED, isCineEnabled);
+    }, 0);
   }
-  if (setCineImplementation) {
-    serviceImplementation._setCine = setCineImplementation;
+
+  public playClip(element, playClipOptions) {
```

Comment:
Given that you are tweaking this, could you remove the level of indirection, and make these all:
public playClip = (element,playClipOptions) => { throw new UnsupportedOperationException('playClip'); }

Reply:
I'm not following this one, can you please elaborate?

Reply:
Just that you can assign methods to the implementation directly, but given that it doesn't work, I would leave it.